language_service/src/qsc_utils.rs

```diff
@@ -69,3 +70,27 @@ pub(crate) fn find_item<'a>(
     };
     (package.items.get(id.item), Some(package))
 }
+
+pub(crate) struct AstIdentFinder<'a> {
```

Comment:
I think it would be a cleaner interface if you exposed a `find_ident()` function instead of this struct. I see you use `AstIdentFinder` in two places above in the same way. Would make the call site cleaner if they called a simple function instead.

Something like:

```rust
fn find_ident(decl: &CallableDecl, node_id: NodeId) -> Option<Ident> {
  let mut finder = AstIdentFinder {
      node_id,
      ident: None,
  };
  finder.visit_callable_decl(decl);
  finder.ident
}
```

(Not sure if the above compiles, wrote it blind)

Reply:
Added `find_ident` and make the `AstIdentFinder` private.